### ISSUE TITLE
Support of superlu_dist-7.2.0

### ISSUE THREAD

**Issue by sagitter** [sfilippone/amg4psblas#3]
Hi all.

`amg4psblas` is not ready for `superlu_dist-7.2.0` (the 7.2.0 one may be next release version in Fedora):

```
mpicc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -I/usr/include/flexiblas    -DHave_MUMPS_ -I/usr/lib64/gfortran/modules/openmpi/MUMPS-5.4.1  -DHave_UMF_ -I/usr/include/suitesparse -DHave_SLUDist_ -DSLUD_VERSION_7 -I/usr/include/openmpi-x86_64/superlu_dist -DHAVE_METIS_ -I/usr/include -DIPK4 -DLPK4 -DLowerUnderscore -DPtr64Bits  -DMETIS_32 -DMETIS_REAL_32 -c amg_dslud_interface.c -o amg_dslud_interface.o
amg_dslud_interface.c:107:3: error: unknown type name 'LUstruct_t'
  107 |   LUstruct_t *LUstruct;
      |   ^~~~~~~~~~
amg_dslud_interface.c:109:3: error: unknown type name 'ScalePermstruct_t'
  109 |   ScalePermstruct_t *ScalePermstruct;
      |   ^~~~~~~~~~~~~~~~~
amg_dslud_interface.c: In function 'amg_dsludist_fact':
amg_dslud_interface.c:143:5: error: unknown type name 'ScalePermstruct_t'; did you mean 'dScalePermstruct_t'?
  143 |     ScalePermstruct_t *ScalePermstruct;
      |     ^~~~~~~~~~~~~~~~~
      |     dScalePermstruct_t
amg_dslud_interface.c:144:5: error: unknown type name 'LUstruct_t'; did you mean 'dLUstruct_t'?
  144 |     LUstruct_t *LUstruct;
      |     ^~~~~~~~~~
      |     dLUstruct_t
amg_dslud_interface.c:145:5: error: unknown type name 'SOLVEstruct_t'; did you mean 'dSOLVEstruct_t'?
  145 |     SOLVEstruct_t SOLVEstruct;
      |     ^~~~~~~~~~~~~
      |     dSOLVEstruct_t
amg_dslud_interface.c:156:5: error: 'choke_on_me' undeclared (first use in this function)
  156 |     choke_on_me;
      |     ^~~~~~~~~~~
amg_dslud_interface.c:156:5: note: each undeclared identifier is reported only once for each function it appears in
amg_dslud_interface.c:182:24: error: 'ScalePermstruct_t' undeclared (first use in this function); did you mean 'dScalePermstruct_t'?
  182 |     ScalePermstruct = (ScalePermstruct_t *) SUPERLU_MALLOC(sizeof(ScalePermstruct_t));
      |                        ^~~~~~~~~~~~~~~~~
      |                        dScalePermstruct_t
amg_dslud_interface.c:182:43: error: expected expression before ')' token
  182 |     ScalePermstruct = (ScalePermstruct_t *) SUPERLU_MALLOC(sizeof(ScalePermstruct_t));
      |                                           ^
amg_dslud_interface.c:183:17: error: 'LUstruct_t' undeclared (first use in this function); did you mean 'dLUstruct_t'?
  183 |     LUstruct = (LUstruct_t *) SUPERLU_MALLOC(sizeof(LUstruct_t));
      |                 ^~~~~~~~~~
      |                 dLUstruct_t
amg_dslud_interface.c:183:29: error: expected expression before ')' token
  183 |     LUstruct = (LUstruct_t *) SUPERLU_MALLOC(sizeof(LUstruct_t));
      |                             ^
amg_dslud_interface.c:184:5: warning: implicit declaration of function 'ScalePermstructInit'; did you mean 'dScalePermstructInit'? [-Wimplicit-function-declaration]
  184 |     ScalePermstructInit(n,n, ScalePermstruct);
      |     ^~~~~~~~~~~~~~~~~~~
      |     dScalePermstructInit
amg_dslud_interface.c:197:31: error: 'options' undeclared (first use in this function)
  197 |     set_default_options_dist(&options);
      |                               ^~~~~~~
amg_dslud_interface.c:201:26: warning: passing argument 3 of 'pdgssvx' from incompatible pointer type [-Wincompatible-pointer-types]
  201 |     pdgssvx(&options, A, ScalePermstruct, b, nl, 0,
      |                          ^~~~~~~~~~~~~~~
      |                          |
      |                          int *
In file included from amg_dslud_interface.c:93:
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:424:22: note: expected 'dScalePermstruct_t *' but argument is of type 'int *'
  424 |                      dScalePermstruct_t *, double *,
      |                      ^~~~~~~~~~~~~~~~~~~~
amg_dslud_interface.c:202:19: warning: passing argument 8 of 'pdgssvx' from incompatible pointer type [-Wincompatible-pointer-types]
  202 |             grid, LUstruct, &SOLVEstruct, berr, &stat, &info);
      |                   ^~~~~~~~
      |                   |
      |                   int *
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:425:46: note: expected 'dLUstruct_t *' but argument is of type 'int *'
  425 |                      int, int, gridinfo_t *, dLUstruct_t *,
      |                                              ^~~~~~~~~~~~~
amg_dslud_interface.c:202:29: warning: passing argument 9 of 'pdgssvx' from incompatible pointer type [-Wincompatible-pointer-types]
  202 |             grid, LUstruct, &SOLVEstruct, berr, &stat, &info);
      |                             ^~~~~~~~~~~~
      |                             |
      |                             int *
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:426:22: note: expected 'dSOLVEstruct_t *' but argument is of type 'int *'
  426 |                      dSOLVEstruct_t *, double *, SuperLUStat_t *, int *);
      |                      ^~~~~~~~~~~~~~~~
amg_dslud_interface.c:213:31: warning: passing argument 2 of 'dSolveFinalize' from incompatible pointer type [-Wincompatible-pointer-types]
  213 |       dSolveFinalize(&options,&SOLVEstruct);
      |                               ^~~~~~~~~~~~
      |                               |
      |                               int *
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:430:54: note: expected 'dSOLVEstruct_t *' but argument is of type 'int *'
  430 | extern void dSolveFinalize(superlu_dist_options_t *, dSOLVEstruct_t *);
      |                                                      ^~~~~~~~~~~~~~~~
amg_dslud_interface.c:162:13: warning: unused variable 'ival' [-Wunused-variable]
  162 |     double *ival;
      |             ^~~~
amg_dslud_interface.c:161:16: warning: unused variable 'irpt' [-Wunused-variable]
  161 |     int *icol,*irpt;
      |                ^~~~
amg_dslud_interface.c:161:10: warning: unused variable 'icol' [-Wunused-variable]
  161 |     int *icol,*irpt;
      |          ^~~~
amg_dslud_interface.c:150:14: warning: unused variable 'drop_tol' [-Wunused-variable]
  150 |     double   drop_tol = 0.0, b[1], berr[1];
      |              ^~~~~~~~
amg_dslud_interface.c:149:14: warning: variable 'trans' set but not used [-Wunused-but-set-variable]
  149 |     trans_t  trans;
      |              ^~~~~
amg_dslud_interface.c:148:41: warning: unused variable 'relax' [-Wunused-variable]
  148 |     int      i, panel_size, permc_spec, relax, info;
      |                                         ^~~~~
amg_dslud_interface.c:148:29: warning: unused variable 'permc_spec' [-Wunused-variable]
  148 |     int      i, panel_size, permc_spec, relax, info;
      |                             ^~~~~~~~~~
amg_dslud_interface.c:148:17: warning: unused variable 'panel_size' [-Wunused-variable]
  148 |     int      i, panel_size, permc_spec, relax, info;
      |                 ^~~~~~~~~~
amg_dslud_interface.c:148:14: warning: unused variable 'i' [-Wunused-variable]
  148 |     int      i, panel_size, permc_spec, relax, info;
      |              ^
amg_dslud_interface.c:136:19: warning: unused variable 'Astore' [-Wunused-variable]
  136 |     NRformat_loc *Astore;
      |                   ^~~~~~
amg_dslud_interface.c: In function 'amg_dsludist_solve':
amg_dslud_interface.c:253:5: error: unknown type name 'ScalePermstruct_t'; did you mean 'dScalePermstruct_t'?
  253 |     ScalePermstruct_t *ScalePermstruct;
      |     ^~~~~~~~~~~~~~~~~
      |     dScalePermstruct_t
amg_dslud_interface.c:254:5: error: unknown type name 'LUstruct_t'; did you mean 'dLUstruct_t'?
  254 |     LUstruct_t *LUstruct;
      |     ^~~~~~~~~~
      |     dLUstruct_t
amg_dslud_interface.c:255:5: error: unknown type name 'SOLVEstruct_t'; did you mean 'dSOLVEstruct_t'?
  255 |     SOLVEstruct_t SOLVEstruct;
      |     ^~~~~~~~~~~~~
      |     dSOLVEstruct_t
amg_dslud_interface.c:267:5: error: 'choke_on_me' undeclared (first use in this function)
  267 |     choke_on_me;
      |     ^~~~~~~~~~~
amg_dslud_interface.c:301:31: error: 'options' undeclared (first use in this function)
  301 |     set_default_options_dist(&options);
      |                               ^~~~~~~
amg_dslud_interface.c:306:26: warning: passing argument 3 of 'pdgssvx' from incompatible pointer type [-Wincompatible-pointer-types]
  306 |     pdgssvx(&options, A, ScalePermstruct, b, ldb, nrhs,
      |                          ^~~~~~~~~~~~~~~
      |                          |
      |                          int *
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:424:22: note: expected 'dScalePermstruct_t *' but argument is of type 'int *'
  424 |                      dScalePermstruct_t *, double *,
      |                      ^~~~~~~~~~~~~~~~~~~~
amg_dslud_interface.c:307:19: warning: passing argument 8 of 'pdgssvx' from incompatible pointer type [-Wincompatible-pointer-types]
  307 |             grid, LUstruct, &SOLVEstruct, berr, &stat, &info);
      |                   ^~~~~~~~
      |                   |
      |                   int *
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:425:46: note: expected 'dLUstruct_t *' but argument is of type 'int *'
  425 |                      int, int, gridinfo_t *, dLUstruct_t *,
      |                                              ^~~~~~~~~~~~~
amg_dslud_interface.c:307:29: warning: passing argument 9 of 'pdgssvx' from incompatible pointer type [-Wincompatible-pointer-types]
  307 |             grid, LUstruct, &SOLVEstruct, berr, &stat, &info);
      |                             ^~~~~~~~~~~~
      |                             |
      |                             int *
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:426:22: note: expected 'dSOLVEstruct_t *' but argument is of type 'int *'
  426 |                      dSOLVEstruct_t *, double *, SuperLUStat_t *, int *);
      |                      ^~~~~~~~~~~~~~~~
amg_dslud_interface.c:311:31: warning: passing argument 2 of 'dSolveFinalize' from incompatible pointer type [-Wincompatible-pointer-types]
  311 |       dSolveFinalize(&options,&SOLVEstruct);
      |                               ^~~~~~~~~~~~
      |                               |
      |                               int *
/usr/include/openmpi-x86_64/superlu_dist/superlu_ddefs.h:430:54: note: expected 'dSOLVEstruct_t *' but argument is of type 'int *'
  430 | extern void dSolveFinalize(superlu_dist_options_t *, dSOLVEstruct_t *);
      |                                                      ^~~~~~~~~~~~~~~~
amg_dslud_interface.c:260:14: warning: unused variable 'drop_tol' [-Wunused-variable]
  260 |     double   drop_tol = 0.0;
      |              ^~~~~~~~
amg_dslud_interface.c:259:14: warning: variable 'trans' set but not used [-Wunused-but-set-variable]
  259 |     trans_t  trans;
      |              ^~~~~
amg_dslud_interface.c:258:41: warning: unused variable 'relax' [-Wunused-variable]
  258 |     int      i, panel_size, permc_spec, relax, info;
      |                                         ^~~~~
amg_dslud_interface.c:258:29: warning: unused variable 'permc_spec' [-Wunused-variable]
  258 |     int      i, panel_size, permc_spec, relax, info;
      |                             ^~~~~~~~~~
amg_dslud_interface.c:258:17: warning: unused variable 'panel_size' [-Wunused-variable]
  258 |     int      i, panel_size, permc_spec, relax, info;
      |                 ^~~~~~~~~~
amg_dslud_interface.c:258:14: warning: unused variable 'i' [-Wunused-variable]
  258 |     int      i, panel_size, permc_spec, relax, info;
      |              ^
amg_dslud_interface.c: In function 'amg_dsludist_free':
amg_dslud_interface.c:339:5: error: unknown type name 'ScalePermstruct_t'; did you mean 'dScalePermstruct_t'?
  339 |     ScalePermstruct_t *ScalePermstruct;
      |     ^~~~~~~~~~~~~~~~~
      |     dScalePermstruct_t
amg_dslud_interface.c:340:5: error: unknown type name 'LUstruct_t'; did you mean 'dLUstruct_t'?
  340 |     LUstruct_t *LUstruct;
      |     ^~~~~~~~~~
      |     dLUstruct_t
amg_dslud_interface.c:341:5: error: unknown type name 'SOLVEstruct_t'; did you mean 'dSOLVEstruct_t'?
  341 |     SOLVEstruct_t SOLVEstruct;
      |     ^~~~~~~~~~~~~
      |     dSOLVEstruct_t
amg_dslud_interface.c:353:5: error: 'choke_on_me' undeclared (first use in this function)
  353 |     choke_on_me;
      |     ^~~~~~~~~~~
amg_dslud_interface.c:375:5: warning: implicit declaration of function 'ScalePermstructFree'; did you mean 'dScalePermstructFree'? [-Wimplicit-function-declaration]
  375 |     ScalePermstructFree(ScalePermstruct);
      |     ^~~~~~~~~~~~~~~~~~~
      |     dScalePermstructFree
amg_dslud_interface.c:376:5: warning: implicit declaration of function 'LUstructFree'; did you mean 'dLUstructFree'? [-Wimplicit-function-declaration]
  376 |     LUstructFree(LUstruct);
      |     ^~~~~~~~~~~~
      |     dLUstructFree
amg_dslud_interface.c:355:19: warning: unused variable 'stat' [-Wunused-variable]
  355 |     SuperLUStat_t stat;
      |                   ^~~~
amg_dslud_interface.c:347:13: warning: unused variable 'berr' [-Wunused-variable]
  347 |     double *berr;
      |             ^~~~
amg_dslud_interface.c:346:14: warning: unused variable 'drop_tol' [-Wunused-variable]
  346 |     double   drop_tol = 0.0;
      |              ^~~~~~~~
amg_dslud_interface.c:345:14: warning: unused variable 'trans' [-Wunused-variable]
  345 |     trans_t  trans;
      |              ^~~~~
amg_dslud_interface.c:344:41: warning: unused variable 'relax' [-Wunused-variable]
  344 |     int      i, panel_size, permc_spec, relax;
      |                                         ^~~~~
amg_dslud_interface.c:344:29: warning: unused variable 'permc_spec' [-Wunused-variable]
  344 |     int      i, panel_size, permc_spec, relax;
      |                             ^~~~~~~~~~
amg_dslud_interface.c:344:17: warning: unused variable 'panel_size' [-Wunused-variable]
  344 |     int      i, panel_size, permc_spec, relax;
      |                 ^~~~~~~~~~
amg_dslud_interface.c:344:14: warning: unused variable 'i' [-Wunused-variable]
  344 |     int      i, panel_size, permc_spec, relax;
      |              ^
amg_dslud_interface.c:341:19: warning: unused variable 'SOLVEstruct' [-Wunused-variable]
  341 |     SOLVEstruct_t SOLVEstruct;
      |                   ^~~~~~~~~~~
amg_dslud_interface.c:333:18: warning: variable 'A' set but not used [-Wunused-but-set-variable]
  333 |     SuperMatrix *A;
      |                  ^
make[2]: *** [../../Make.inc:116: amg_dslud_interface.o] Error 1
```

**Comment by sfilippone** [comment]
Ok, will need to look into the new SuperLU_Dist interfaces and add a configry check for version.

**Comment by sfilippone** [comment]
Could you please try compiling against branch fix-sludist7 ? 
If it works, I'll merge into development and master
Thanks

**Comment by sagitter** [comment]
I used [this patch](https://github.com/sfilippone/amg4psblas/compare/fix-sludist7.patch)
Build log: https://download.copr.fedorainfracloud.org/results/sagitter/ForTesting/fedora-rawhide-x86_64/04084803-amg4psblas/builder-live.log.gz

**Comment by sfilippone** [comment]
Looks like you simply disabled SuperLU_Dist compilation. Are you able to test with the other branch? 
It should work, but right now I have no time to download and install.
Otherwse, I can merge into development and create a new tag

**Comment by sagitter** [comment]
>Looks like you simply disabled SuperLU_Dist compilation. Are you able to test with the other branch?

There are two builds: the first one is serial against SuperLU, the second one is MPI against superlu_dist-7.2.0
Search the "Welcome to the AMG4PSBLAS 1.0.0 configure Script." line to find them easily.

**Comment by sagitter** [comment]
Fixed in e02df3725eae763ac3727b86bc528beaab49e4c5
